search2/python/search/utils/beanstalkd.py

```diff
@@ -0,0 +1,155 @@
+"""TODO: Add description here"""
+from glob import glob
+from os import path
+import time
+import traceback
+from typing import Any, NamedTuple, List, Callable, Optional
+
+import asyncio
+from orjson import loads, dumps  # pylint: disable=no-name-in-module
+from pystalk import BeanstalkClient, BeanstalkError  # type: ignore
+from opensearchpy.exceptions import NotFoundError
+
+class Message(NamedTuple):
```

Comment:
These NamedTuple classes look great, I just wouldn't defer the docstrings

Reply:
We should discuss expectations for docstrings. They are not needed for many of these small, descriptively named methods. Pylint requires them, and I propose we drop that requirement at the linter level.

Reply:
I agree that the English-language meanings of the class names are clear.  I don't think it's as clear from the names what they represent in the context of this code.  For example, Publisher has a host, port, queue and message.  Host and port are clearly enough an address, but what does the queue str represent?  Why does every Publisher have its own Message?  I offer these as examples of questions a good-faith reader might have when reading this code.

It might make sense to disable the docstring requirement for private methods, but would recommend against dropping it for public-facing methods.  Happy to discuss further offline.